woocommerce/google-listings-and-ads
# Add DB migration and remove an unneccesary index from merchant_issues table

### Changes proposed in this Pull Request:



This PR adds a database migration method so that we can apply several DB queries whenever the extension is updated.

It also removes the `product_issue` unique index from the `merchant_issues` table because it's not used and in some cases it was causing an error for users (http://wordpress.org/support/topic/fatal-error-upon-update-1-4-0-specified-key-was-too-long).

We already make sure that the product issues are unique by applying an md5 hash of the issues so the unique key was no longer necessary and removing it seemed like a quick solution to the DB problem mentioned in the forum.

I added unit tests for the `Migrator` class and part of the `DBHelper` class. It can be a bit tricky when adding unit tests that deal with raw database features such as creating and dropping tables. For example, when creating a table to test the `DBHelper::table_exists` method, WordPress forcefully created a `TEMPORARY` table, which meant that the method would always return `false`.

### Detailed test instructions:

1. Change the `WC_GLA_VERSION` constant in the main plugin file to `1.4.1`
2. Reload the WP admin
3. Confirm that the `product_issue` index is now removed from the `merchant_issues` table


### Changelog entry

> Fix - Database error: "Specified key was too long". i.e. removed the `product_issue` index from the `merchant_issues` table.
